pretty_midi/pretty_midi.py

```diff
@@ -117,6 +146,38 @@ def _load_tempo_changes(self, midi_data):
                     if tick_scale != last_tick_scale:
                         self.__tick_scales.append((event.tick, tick_scale))
 
+    def _load_metadata(self, midi_data):
+        """Populates the lists of tempo key and time signature changes
+
+        Populates self.__time_signatures with TimeSignature objects.
+            It can be accessed by the get_time_signatures method.
+        Populates self.__key_changes with KeySignature objects.
+            It can be accessed by the get_key_signatures method.
+
+        Parameters
+        ----------
+            - midi_data : midi.FileReader
+                MIDI object from which data will be read
+        """
+
+        # list to store key signature changes
+        self.__key_changes = []
```

Comment:
> If the midi file is a transcription from someone's performance on an instrument, then the user might have the need to change it.

Yes, `pretty_midi` is meant to handle the case for MIDI file creation as well as reading/analyzing/modification, so I think we need to provide a way for these lists to be modified.  Currently, users can put whatever they want into, for example, the `.instruments` list (and the `Instrument.notes` list, etc).  We could provide setters which do sanity checks, but the Python way is usually just to provide things publicly for people to do the sane thing with, rather than use private members and setters/getters as in, say, C++.